lumen/ai/agents.py

```diff
@@ -578,9 +578,9 @@ async def _create_valid_sql(
                     sql_query = sql_query.replace(a_table, renamed_table)
                 else:
                     renamed_table = a_table
+                # Remove source prefixes from table names
                 if SOURCE_TABLE_SEPARATOR in renamed_table:
-                    # Remove source prefixes from table names
-                    renamed_table = re.sub(rf".*?{SOURCE_TABLE_SEPARATOR}", "", renamed_table)
+                    _, renamed_table = renamed_table.split(SOURCE_TABLE_SEPARATOR)
```

Comment:
For maximum stability, I think we should do maxsplit=1.

Reply:
Sure.